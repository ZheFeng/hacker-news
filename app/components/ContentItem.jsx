import React, { PropTypes } from 'react';
import { Map } from 'immutable';

const codeStyle = {
  fontSize: '10px',
  color: '#ddd',
  display: 'none',
};

function dateDisplay(time) {
  const d = new Date(time);
  return d.toDateString();
}

const ContentItem = ({ data }) => (
  <div className="card" style={{ margin: '10px' }}>
    <div className="card-block">
      <h4 className="card-title">{data.get('title')}</h4>
      <h6 className="card-subtitle mb-2 text-muted">{dateDisplay(data.get('time'))}</h6>
      <p className="card-text">{data.get('text')}</p>
      <a
        href="{data.get('url')}"
        className="card-link"
        target="_blank"
        rel="noopener noreferrer"
      >
        {data.get('url')}
      </a>
      <pre style={codeStyle} className="pre-scrollable">
        <code>
          {JSON.stringify(data.toObject(), null, ' ')}
        </code>
      </pre>
    </div>
  </div>
);

ContentItem.propTypes = {
  data: PropTypes.instanceOf(Map).isRequired,
};

export default ContentItem;
