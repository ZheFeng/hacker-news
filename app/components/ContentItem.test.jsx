/* eslint-disable import/no-extraneous-dependencies, max-nested-callbacks */
/* global describe, it, expect */
import React from 'react';
import renderer from 'react-test-renderer';

import ContentItem from './ContentItem';

const title = 'test title';
const d = new Date(2010, 1, 1);
const time = d.getTime();
const text = 'test text';
const url = 'http://test.com';

describe('<ContentItem />', () => {
  it('Snapshot Test', () => {
    const component = renderer.create(
      <ContentItem title={title} time={time} text={text} url={url} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
