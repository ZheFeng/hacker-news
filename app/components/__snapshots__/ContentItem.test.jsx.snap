exports[`<ContentItem /> Snapshot Test 1`] = `
<div
  className="card"
  style={
    Object {
      "margin": "10px",
    }
  }>
  <div
    className="card-block">
    <h4
      className="card-title">
      test title
    </h4>
    <h6
      className="card-subtitle mb-2 text-muted">
      Mon Feb 01 2010
    </h6>
    <p
      className="card-text">
      test text
    </p>
    <a
      className="card-link"
      href="http://test.com"
      rel="noopener noreferrer"
      target="_blank">
      http://test.com
    </a>
  </div>
</div>
`;
